docs/3.2. Discovery - Peer to Peer Operation.md

```diff
@@ -15,6 +15,41 @@ The following pre-requisites are optional aspects of the [Node API](../APIs/Node
 
 Both of the above are properties of the Node API and require no additional implementation steps.
 
+## DNS-SD Advertisement
+
+Node APIs MAY produce an mDNS advertisement of the type \_nmos-node.\_tcp in order to support peer to peer operation. This SHOULD NOT be advertised when a Registration API is present on the network, or when peer to peer operation is disabled.
+
+The IP address and port of the Node API MUST be identified via the DNS-SD advertisement, with the full HTTP path then being resolved via the standard NMOS API path documentation.
+
+Multiple DNS-SD advertisements for the same API are permitted where the API is exposed via multiple ports and/or protocols.
+
+## DNS-SD TXT records
+
+**api\_proto**
+
+The DNS-SD advertisement MUST be accompanied by a TXT record of name 'api\_proto' with a value of either 'http' or 'https' (lower-case) dependent on the protocol in use by the Node API web server.
+
+**api\_ver**
+
+The DNS-SD advertisement MUST be accompanied by a TXT record of name 'api\_ver'. The value of this TXT record is a comma separated list of API versions supported by the server. For example: 'v1.0,v1.1,v2.0'. There should be no whitespace between commas, and versions should be listed in ascending order.
+
+**ver\_**
+
+When a Node is operating in peer to peer mode it MUST additionally advertise the following mDNS TXT records as part of its Node advertisement. If a Node is successfully registered with a Registration API it MUST withdraw advertisements of these TXT records. There is no requirement to register these TXT records with a unicast DNS service.
+
+| **TXT Record Name** | **Corresponding Node API Resource** |
+|---------------------|-------------------------------------|
+| ver\_slf            | self                                |
+| ver\_src            | sources                             |
+| ver\_flw            | flows                               |
+| ver\_dvc            | devices                             |
+| ver\_snd            | senders                             |
+| ver\_rcv            | receivers                           |
+
+The value of each of the above should be an unsigned 8-bit integer initialised to '0'. This integer MUST be incremented and mDNS TXT record updated whenever a change is made to the corresponding HTTP API resource on the Node. The integer must wrap back to a value of '0' after reaching a maximum value of '255' (MAX_UINT8_T).
+
+For example, the 'ver_src' TXT record must be created when the Node first advertises itself via mDNS. If the data held within the HTTP resource for /sources is added to, removed from or edited, then the 'ver_src' text record must be modified (value incremented).
+
 ## Querying
```

Comment:
Just noting that above changed 'Querying' heading to 'Client Interaction Procedure'.